src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -304,7 +304,7 @@ public void setPeriod(String period) {
     @Override
     public void visitJavadocToken(DetailNode ast) {
         final Optional<DetailNode> inlineSummaryTag = getInlineSummaryTag(ast);
-        if (inlineSummaryTag.isPresent()) {
+        if (inlineSummaryTag.isPresent() && isDefinedFirst(inlineSummaryTag.get())) {
```

Comment:
Is it better to update getInlineSummaryTag(ast) to return nothing if summary tag is not first ?

Reply:
I feel it would be better to keep these methods separate, the first one just gets the inline summary tag, and the second one checks whether it is a suitable candidate for getting into the summary. 
This makes a clear distinction between just having a summary tag and it being applicable for getting into the actual summary part.

Reply:
If we do validation of incorrectly placed tag , I would agree with you, but we just need to search for tag in first place only, if not found, it is not what we need ( no matter where it is located).

@strkkk , can we ask for second opinion from you?

Reply:
> If we do validation of incorrectly placed tag , I would agree with you, but we just need to search for tag in first place only, if not found, it is not what we need ( no matter where it is located).

Hmm, now I think about it, I feel that would be a better design (More abstract and easier to debug and manage), I changed to it.

Reply:
I see now there are 2 methods, looks fine for me. 

I suggest to change unnecessary if to one more step in optional transformation. I mean, instead of 
```
        if (node.isPresent() && (!isSummaryTag(node.get()) || !isDefinedFirst(node.get()))) {
            node = Optional.empty();
        }
```
you can just do 
```
return Arrays.stream(javadoc.getChildren())
                .filter(SummaryJavadocCheck::isInlineTagPresent)
                .findFirst()
                .map(SummaryJavadocCheck::getInlineTagNodeWithinHtmlElement);
.filter(node -> !isSummaryTag(node) || !isDefinedFirst(node));
```
this will be easier

Reply:
> this will be easier

The condition won't be quite the same but good observation. Changed it.